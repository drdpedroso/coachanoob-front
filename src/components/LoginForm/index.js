import React, { useState } from 'react'
import PropTypes from 'prop-types';
import Button from '../Button'
import { Form } from 'semantic-ui-react'

const LoginForm = (props) => {
    const [form, setForm] = useState({})

    const onChangeInputValue = (e, key) => {
        const { value } = e.target
        setForm({
            ...form,
            [key]: value
        })
    }

    const onSubmit = () => {
        return typeof props.onSubmit === 'function' ? props.onSubmit(form) : null
    }

    const shouldDisableButton = () => {
        return !(form.email && form.password)
    }

    return (
        <Form>
            <Form.Field>
                <label>E-mail</label>
                <input data-testid="email" placeholder='E-mail' type="email" onChange={(e) => onChangeInputValue(e, 'email')}/>
            </Form.Field>
            <Form.Field>
                <label>Senha</label>
                <input data-testid="password" placeholder='Senha' type="password" onChange={(e) => onChangeInputValue(e, 'password')}/>
            </Form.Field>
            <Button data-testid="submit" type='submit' fluid disabled={shouldDisableButton()} onClick={onSubmit}>Entrar</Button>
        </Form>
    )
};

LoginForm.propTypes = {
    onSubmit: PropTypes.func
};

export default LoginForm
