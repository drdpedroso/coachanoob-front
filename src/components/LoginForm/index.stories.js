import React from 'react';
import {storiesOf} from '@storybook/react';
import {action} from '@storybook/addon-actions';

import LoginForm from './';

storiesOf('LoginForm', module)
    .add('with default', () => (
        <LoginForm onSubmit={(val) => action(val)} />
    ))

