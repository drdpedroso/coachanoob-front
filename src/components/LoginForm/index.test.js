import React from 'react';
import LoginForm from './index'
import renderer from 'react-test-renderer'
import { render, fireEvent, cleanup } from '@testing-library/react'

afterEach(cleanup)

describe('LoginForm', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<LoginForm />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should send login form information to onSubmit button', () => {
        const onSubmit = jest.fn()
        const emailValue = 'drdpedroso@gmail.com'
        const passwordValue = 'senha123'
        const component = render(<LoginForm onSubmit={onSubmit} />)
        const email = component.getByTestId('email')
        const password = component.getByTestId('password')
        const submitButton = component.getByTestId('submit')
        fireEvent.change(email, {target: {value: emailValue}})
        fireEvent.change(password, {target: {value: passwordValue}})
        fireEvent.click(submitButton)
        expect(email.value).toBe(emailValue)
        expect(password.value).toBe(passwordValue)
        expect(onSubmit).toBeCalledWith(
            {
                email: emailValue,
                password: passwordValue
            }
        )
    })

    it('should have submit button disabled if no form is filled', () => {
        const onSubmit = jest.fn()
        const component = render(<LoginForm onSubmit={onSubmit} />)
        const submitButton = component.getByTestId('submit')
        fireEvent.click(submitButton)
        expect(onSubmit).not.toBeCalled()
    })

    it('should have submit button disabled if only email is filled', () => {
        const onSubmit = jest.fn()
        const emailValue = 'drdpedroso@gmail.com'
        const component = render(<LoginForm onSubmit={onSubmit} />)
        const email = component.getByTestId('email')
        fireEvent.change(email, {target: {value: emailValue}})
        const submitButton = component.getByTestId('submit')
        fireEvent.click(submitButton)
        expect(onSubmit).not.toBeCalled()
    })

    it('should show error message when error object is passed', () => {
        const error = {content: 'Error content message', header: 'Something went worng'}
        const onSubmit = jest.fn()
        const component = render(<LoginForm onSubmit={onSubmit} error={error}/>)
        const errorMessage = component.getByTestId('error-message')
        expect(onSubmit).not.toBeCalled()
        expect(errorMessage).toBeDefined()
    })

})
