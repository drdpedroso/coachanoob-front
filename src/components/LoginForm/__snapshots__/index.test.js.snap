// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <label>
      E-mail
    </label>
    <input
      data-testid="email"
      onChange={[Function]}
      placeholder="E-mail"
      type="email"
    />
  </div>
  <div
    className="field"
  >
    <label>
      Senha
    </label>
    <input
      data-testid="password"
      onChange={[Function]}
      placeholder="Senha"
      type="password"
    />
  </div>
  <button
    className="ui fluid disabled button"
    data-testid="submit"
    disabled={true}
    onClick={[Function]}
    tabIndex={-1}
    type="submit"
  >
    Entrar
  </button>
</form>
`;
